Miscellaneous http-service features to support a future web-ui results view

drothlis opened the pull request:
Extracted from #25. These are ready to merge now whereas #25 is going to take a while to get the results view up to par with the current static view.

wmanley commented:
> http-service /results endpoint: Add count and limit parameters

It's also worth mentioning that the `category` is now passed in by the query string, rather than as part of the `path` of the URL.  I guess this is probably the most important part.

Also there are no tests that actually check that any of the query parameters have any effect.  Maybe it's worth removing them rather than migrating them.

wmanley commented on tests/test_http_service.py:
```diff
@@ -691,11 +691,25 @@ def test_that_a_completed_result_can_be_retrieved_via_results_interface(runner):
+         'tests/example_tests.py::test_sleeping'])
+    assert wait_until(lambda: get_generation() == prev_generation + 1)
+    assert wait_until(lambda: get_generation() == prev_generation + 2)
+
```
Nice test.

drothlis commented:
> It's also worth mentioning that the category is now passed in by the query string, rather than as part of the path of the URL. I guess this is probably the most important part.

I updated the commit message.

> Also there are no tests that actually check that any of the query parameters have any effect. Maybe it's worth removing them rather than migrating them.

I added tests.
